client/src/components/SignUpForm.js

```diff
@@ -55,6 +57,8 @@ export default function SignUp() {
                                 setSeverity("success");
                                 setMessage("Successfully registered");
                                 setOpen(true);
+                                user.setProfile(data.user)
```

Comment:
Same here, let's move this into userContext

Reply:
Refactored to have this in the UserContextProvider component